### PR TITLE
AJ-448 first pass at adding swagger codegen

### PR DESCRIPTION
**Pull request by calypsomatic**
Running `./gradlew build` will now generate code from the swagger doc and put it in `/client`.  Note this folder and thus the generated code is .gitignored.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -45,3 +47,18 @@ tasks.named('test') {
     }
 }
 
+swaggerSources {
+    server {
+        inputFile = file('static/swagger/openapi-docs.yaml')
+        code {
+            language = 'java'
```
Comment (calypsomatic) [inline build.gradle:54]: Do we want 'java' or 'spring' here?  'spring' gives more readable java files, and I see it used in other repos, but which form does the client want it in?

Comment (davidangb) [reply build.gradle:54]: on `java` vs `spring`: according to the listings at https://github.com/swagger-api/swagger-codegen, `java` generates a client, while `spring` generates server stubs. So, we definitely want `java`

File: build.gradle [hunk 2]
```diff
@@ -45,3 +47,18 @@ tasks.named('test') {
...
+            jvmArgs = ['--add-opens=java.base/java.util=ALL-UNNAMED'] // for Swagger Codegen v3 on Java 16+
```
Comment (davidangb) [inline build.gradle:55]: what does this do? I don't immediately get an error without this, but I haven't tried using the generated files. Is it necessary?

File: build.gradle [hunk 3]
```diff
@@ -45,3 +47,18 @@ tasks.named('test') {
     }
 }
 
+swaggerSources {
+    server {
```
Comment (davidangb) [inline build.gradle:51]: I think this needs to be `client`, not `server`

**Review by davidangb — COMMENTED** [review]
Some initial config thoughts:
* I think we need config for the java package names; else it looks like the generated files are in `io.swagger.client`. See https://github.com/DataBiosphere/jade-data-repo/blob/0ae97efce6deb81a41001c4f245e217d768ce0c9/datarepo-client/build.gradle#L59-L62 for an example.
* should we add an `outputDir` config setting? I think this would make it easier for developers to be able to read the build.gradle and immediately know where the generated files are created. See https://github.com/DataBiosphere/jade-data-repo/blob/0ae97efce6deb81a41001c4f245e217d768ce0c9/datarepo-client/build.gradle#L57 for an example. If this creates a new directory, make sure to add it to `.gitignore` so we don't commit generated files.
* from reading the generated files, it looks like by default the Java client uses okhttp/gson. I see that Data Repo and WSM, and CBAS are using jersey2, should we use that? Example: https://github.com/DataBiosphere/cbas/blob/5d7528b2e2bef3a834938d3eb73051885873764e/client/swagger.gradle#L17; list of Java engine options: https://github.com/swagger-api/swagger-codegen#modifying-the-client-library-format
* from reading the generated files, it is creating tests for the apiclient. We might want to skip generating those; example https://github.com/DataBiosphere/cbas/blob/5d7528b2e2bef3a834938d3eb73051885873764e/client/swagger.gradle#L23-L26

File: build.gradle (hunk 2 above)
Comment (calypsomatic) [reply build.gradle:55]: The docs said it was necessary for Java 16+.

**Pull request closed without merging**